artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerReceiverContext.java

```diff
@@ -60,6 +61,8 @@ public class ProtonServerReceiverContext extends ProtonAbstractReceiver {
 
    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
+   private static final AtomicLong ID_GENERATOR = new AtomicLong();
```

Comment:
I was actually suggesting that the broker allocated a differentiating long ID to _all_ producers for consistency, rather than just the AMQP ones, in similar manner as it does for every message received.

Reply:
Both AMQ and Openwire create this on the client side so tracking it at the broker wouldnt work. They both use UUIDs. This is how the producers are created and closed by both.

Reply:
The brokers own differentiating ID wouldnt need to be the same as the clients string id though. It would just be a broker-side conceptual id. Exactly the same way broker assigns every message a long id for its own purposes, even though messages often already include client-generated MessageID strings, and individual protocols may have their own specific independent way of referencing a particular message for acks etc.

Reply:
i don't really understand what you mean, the client allocates an ID that it uses to control the creation and deletion of the producer and then adds this to the method so the broker knows what producer is being sent to. I dont see how th ebroker can create its own id

Reply:
ok, so I understand and I have implemented it as you suggested